### ISSUE TITLE
Host `env` randomly breaks

### ISSUE THREAD

**Issue by reyemxela** [1player/host-spawn#34]
First off, I'm honestly not sure if this is actually a host-spawn issue, an issue with the flatpak helper, or something dumb with my setup.

**The main issue:** occasionally when run from a distrobox, `host-spawn env` returns minimal entries. No `DISPLAY`/`WAYLAND_DISPLAY`, no `XDG_SESSION_*`, etc. You can see both `env`s in the attached straces.

**Context:** I have a few distrobox containers that I use for most of my day-to-day stuff, with several aliases and symlinks for running host commands (podman, flatpak, vscode, etc.). Everything works great for the most part, until it suddenly breaks again. It took a while to narrow down what the actual issue was, the main problems I would have were GUI apps (like vscode) not wanting to launch because they couldn't find `DISPLAY`. And `env` within the distrobox itself shows up fine, so running GUI apps installed within the distrobox works. It's only when trying to `host-spawn` something that I have this issue.

It's also a tough issue to diagnose because I can't figure out for sure how to trigger it to break, or how to fix it once it's broken. Deleting/recreating the distrobox doesn't usually help, though I did see it fix itself earlier after deleting the distrobox, fully logging out of my Plasma session, then logging back in and recreating the distrobox.

Doing a diff on the strace logs, the thing that stuck out the most to me is the `epoll_pwait` on line 193 in both files. From that point, things go in fairly different directions.

Like I said, not sure it's even an issue with host-spawn itself. It could easily be something dumb I'm doing or missing in my .profile/.bashrc, or something else set up funky on my machines. Just figured I'd ask here to see if you've seen anything like this, or had any ideas of where to look.

[host-spawn-broken.log](https://github.com/1player/host-spawn/files/14872642/host-spawn-broken.log)
[host-spawn-working.log](https://github.com/1player/host-spawn/files/14872644/host-spawn-working.log)

**Comment by 1player** [comment]
Thanks.. I might not have time to look for a few days but as an idea.. can you reproduce this with `flatpak-spawn --host`? We use the same host-side service as flatpak-spawn, so if you can't reproduce it with that tool, it definitely must be a host-spawn issue.

**Comment by reyemxela** [comment]
Yeah, forgot to mention that. `flatpak-spawn --host` does seem to have the same behavior, so it's definitely something on that side of things. Sorry if this isn't a good place to ask about it then.

**Comment by reyemxela** [comment]
Just wanted to report back, I started digging deeper on the flatpak side than I had previously. Looks like it's all based on the env of the `flatpak-session-helper` service when it's first started. If that service gets started when there's no graphical session running, it ends up running under a very minimal environment, and things stay broken. Once I restart that service, things work again.

Obviously that makes logical sense. My main confusion is why nobody else that uses distrobox+host-spawn/flatpak-spawn/etc. seems to have these issues. I'm gonna keep digging on that now that I know what the problem (and fix) is.

But it's not anything really to do with host-spawn, so I'll close this issue. Thanks again for being willing to look into it anyway!
